lib/puppet/type/iis_application_pool.rb

```diff
@@ -1,48 +1,395 @@
+        fail("Invalid value '#{value}'. Valid values are 'HttpLevel', 'TcpLevel'")
+      end
+    end
+  end
+
+  newproperty(:rapid_fail_protection, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+  end
+
+  newproperty(:rapid_fail_protection_interval, :parent => PuppetX::PuppetLabs::IIS::Property::PositiveInteger) do
+    # rapid_fail_protection_interval = (New-TimeSpan -Minutes 5).ToString()
+    # 00:05:00
+    # ([ValidateRange(1, 144000)]$valueInMinutes = [TimeSpan]::Parse($_).TotalMinutes); $?
+    validate do |value|
+      super value
+      fail "#{self.name.to_s} should be less than or equal to 144000" unless value.to_i <= 144000
+    end
+  end
+
+  newproperty(:rapid_fail_protection_max_crashes, :parent => PuppetX::PuppetLabs::IIS::Property::PositiveInteger) do
+    validate do |value|
+      super value
+      fail "#{self.name.to_s} should be less than or equal to 2147483647" unless value.to_i <= 2147483647
+    end
+  end
+
+  newproperty(:auto_shutdown_exe, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+  end
+
+  newproperty(:auto_shutdown_params, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+  end
+
+  newproperty(:disallow_overlapping_rotation, :boolean => true) do
+    newvalue(:true)
+    newvalue(:false)
+       
+    munge do |value|
+      resource.munge_boolean(value)
+    end
+    defaultto false
+  end
+
+  newproperty(:disallow_rotation_on_config_change, :boolean => true) do
+    newvalue(:true)
+    newvalue(:false)
+       
+    munge do |value|
+      resource.munge_boolean(value)
+    end
+    defaultto false
+  end
+
+  newproperty(:log_event_on_recycle, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+  end
+
+  newproperty(:restart_memory_limit, :parent => PuppetX::PuppetLabs::IIS::Property::PositiveInteger) do
+  end
+
+  newproperty(:restart_private_memory_limit, :parent => PuppetX::PuppetLabs::IIS::Property::PositiveInteger) do
+  end
+
+  newproperty(:restart_requests_limit, :parent => PuppetX::PuppetLabs::IIS::Property::PositiveInteger) do
+  end
+
+  newproperty(:restart_time_limit, :parent => PuppetX::PuppetLabs::IIS::Property::PositiveInteger) do
+    # restart_time_limit = (New-TimeSpan -Minutes 5).ToString()
+    # 00:05:00
+    # ([ValidateRange(0, 432000)]$valueInMinutes = [TimeSpan]::Parse($_).TotalMinutes); $?
+    validate do |value|
+      super value
+      fail "#{self.name.to_s} should be less than or equal to 432000" unless value.to_i <= 432000
+    end
+  end
+
+  newproperty(:restart_schedule, :parent => PuppetX::PuppetLabs::IIS::Property::PositiveInteger) do
+    # restart_time_limit = (New-TimeSpan -Minutes 5).ToString()
+    # 00:05:00
+    # ($_ -eq '') -or
+    # (& {
+    #     ([ValidateRange(0, 86399)]$valueInSeconds = [TimeSpan]::Parse($_).TotalSeconds); $?
+    # })
+    validate do |value|
+      super value
+      fail "#{self.name.to_s} should be less than or equal to 86399" unless value.to_i <= 86399
+    end
+  end
```

Comment:
All of the properties should have short doc strings explaining what each is, mentioning allowed values (where `newvalues` is not sufficient), reference to the upstream parameter name (for searching), and the implied unit of the value.

Usually I copy the the summary line from upstream documentation under fair use.